### DISCUSSION TITLE
How to use executable_path like original selenium python

### DISCUSSION THREAD

**Discussion by zaingithub** [seleniumbase/SeleniumBase#1761] [Q&A] [answered]
Hi, how to use executable_path and chrome options like original selenium python

**Comment by mdmintz** [comment, marked as answer]
To set the browser executable path (Chrome binary location), you can use a command-line option when you run ``pytest``:
``--binary-location=PATH``

Example:

```bash
pytest test_demo_site.py --binary-location="/Applications/Google Chrome.app/Contents/MacOS/Google Chrome"
```

The options are similar, but different. (See https://github.com/seleniumbase/SeleniumBase/blob/master/seleniumbase/plugins/pytest_plugin.py)

**Comment by zaingithub** [comment]
Thank you for your reply :)

I don't use pytest. Can I use seleniumbase like regular selenium?

I've created a class that I can access this driver with self.driver, something like this,

```
class MyDriver(object) :

	def chrome_options(self):
		### code here


	def driver_start(self):
		chrome_options			= self.chrome_options()
		desired_capabilities 	= chrome_options.to_capabilities()
		desired_capabilities["acceptInsecureCerts"] = True
		driver = webdriver.Chrome(
			chrome_options=chrome_options, desired_capabilities=desired_capabilities, executable_path=self.path
		)
		return driver


driver_class = MyDriver()
driver_class = driver_class.driver_start()
```

I don't know how to implement Selenium Base like that.

Please help, thank you.

**Reply by mdmintz** [reply]
See https://github.com/seleniumbase/SeleniumBase/blob/master/help_docs/syntax_formats.md and choose one of the syntax formats for structuring your tests.

**Reply by zaingithub** [reply]
Thank you for your response.

The most feasible for me is to use option [#23. The driver manager (via direct import)](https://github.com/seleniumbase/SeleniumBase/blob/master/help_docs/syntax_formats.md#-23-the-driver-manager-via-direct-import)

But the problems I'm having:
- can't use some selenium built-in features like driver.find_elements_by_xpath("//xpath")
- implementation of : chrome_options, and desired_capabilities if I use this direct import Driver

actually I only want the undetected "UC" feature from Seleniumbase,

because selenium alone cannot bypass Google's bot detection.

**Reply by mdmintz** [reply]
Selenium removed ``find_element_by_*`` and ``find_elements_by_*`` methods. See https://stackoverflow.com/a/72754667/7058266 for details and the workaround.

For an undetected script to log into gmail using the ``SB`` manager (https://github.com/seleniumbase/SeleniumBase/blob/master/help_docs/syntax_formats.md#sb_sf_21):

```python
from seleniumbase import SB

with SB(uc=True) as sb:
    sb.open("https://www.google.com/gmail/about/")
    sb.click('a[data-action="sign in"]')
    sb.type('input[type="email"]', "test123@gmail.com")
    sb.click('button:contains("Next")')
    sb.sleep(5)
    # sb.type('input[type="password"]', PASSWORD)
    # sb.click('button:contains("Next")')
```

(As seen in https://stackoverflow.com/a/74384231/7058266)

It's more lines of code to do it via the Driver Manager alone because it doesn't have the smart SeleniumBase APIs, but you can do the same via regular Selenium when using the Driver Manager in ``uc=True`` mode.

**Comment by zaingithub** [comment]
Hi @mdmintz ,

I still want to be able to use seleniumbase like regular selenium, with the implementation of undetected mode (uc).

Alternatively, could you please help me with providing the code implementation of UC for the Driver Manager?

Thank you very much.

**Reply by mdmintz** [reply]
An example of using the ``Driver`` Manager with ``uc`` mode enabled:

```python
from seleniumbase import Driver
from seleniumbase import page_actions

driver = Driver(headless=True, uc=True)
driver.get("https://nowsecure.nl")
page_actions.wait_for_text(driver, "OH YEAH, you passed!", "h1")
print(driver.find_element("css selector", "body").text)
screenshot_name = "now_secure_image.png"
driver.save_screenshot(screenshot_name)
print("\nScreenshot saved to: %s" % screenshot_name)
driver.quit()
```

**Reply by JaveriaButt** [reply]
can you tell me is there any specific reason for using nowsecure.nl before sending request to our target website?

**Reply by mdmintz** [reply]
@JaveriaButt That's just an example. Use any website you want.

**Comment by krypterro** [comment]
Is there a way to specify the path to the binary via the Driver construction?

For example:

```python
driver_version = 127
chrome_path = os.path.join(dir_path, "chrome", "chrome")
driver = Driver(
                    uc=True,
                    headed=False,
                    devtools=False,
                    remote_debug=False,
                    driver_version=chrome_version,
                    binary_location=chrome_path,
                    proxy=f"socks5://{use_proxy}",
                    no_sandbox=True,
                )

```

This works in regular Selenium of course, but it's not working for me in SB for some reason.

**Reply by mdmintz** [reply]
Something like this should work: `binary_location="/Applications/Brave Browser.app/Contents/MacOS/Brave Browser"` if you were going for another Chromium browser (although Edge won't work in UC Mode).

I mentioned that a bit here: https://github.com/seleniumbase/SeleniumBase/discussions/2424#discussioncomment-8100989

And I demoed that in the 2nd UC Mode video tutorial on YouTube at 1521 seconds into the video (25m & 21s):
https://www.youtube.com/watch?v=2pTpBtaE7SQ&t=1521s
